lib/tasks/spec.rake

```diff
@@ -2,4 +2,12 @@ require 'rspec/core/rake_task'
 
 RSpec::Core::RakeTask.new(:spec)
 
-task :default => :spec
+desc "Run Go tests"
+task :go_test do
+  cmd = "go test -v ./..."
+  puts cmd
+  exit $?.exitstatus unless system(cmd, :chdir => File.expand_path("../../manageiq-operator", __dir__))
+  puts
```

Comment:
This can be simplified to just use the `rake` builtin `sh`:

```ruby
task :go_test do
  sh "go test -v ./..."
end
```

Reply:
Okay, I lied a bit here because of the need to do the `:chdir`... let me look up and see if that is supported with `sh`.  Otherwise you could wrap in a `chdir { }`:

```
task :go_test do
  chdir File.expand_path("../../manageiq-operator", __dir__)) do
    sh "go test -v ./..."
  end
end

Reply:
Yeah, it does support `:chdir => path`, because it just wraps `system`:

https://github.com/ruby/rake/blob/5c60da8644a9e4f655e819252e3b6ca77f42b7af/lib/rake/file_utils.rb#L54

Reply:
Does it exit with that same exit code on failure?

Reply:
> Does it exit with that same exit code on failure?

Does that matter?  I don't see why it would, but no, it doesn't:

https://github.com/ruby/rake/blob/5c60da8644a9e4f655e819252e3b6ca77f42b7af/lib/rake/file_utils.rb#L66-L68

Simply calls `fail` and prints the exit status of the command.

Reply:
That's probably fine - my usual pattern for wrapping shell commands tries to exit with the same exit code.  I just couldn't follow the rake code to understand what it was doing on failure.

Reply:
Done.